### DISCUSSION TITLE
Updates not working on Security Onion!

### DISCUSSION THREAD

**Discussion by ITgeevoSOC** [Security-Onion-Solutions/securityonion#5721] [Unsupported Versions]
When we enter the command soc-rule-update it gets stuck during the process and it doesn't finish.

Please find attached the screenshot:

![Capture](https://user-images.githubusercontent.com/90396548/135404211-538237ff-6663-405a-ae62-f8b48b54f7d1.PNG)

Please note that we have 4 virtual CPU and 8 GB RAM.

Kind regards,

**Comment by clairmont32** [comment]
Do you have a firewall preventing internet access from the server? This is seemingly failing at performing the rule download.

**Comment by ITgeevoSOC** [comment]
When we setup security onion,  we had use the same command and downloaded the rules successfully. After few days we had this issue.

Please note that is the 2nd security onion that we had this issue with suricata rule updates.

**Comment by clairmont32** [comment]
Do you mind checking if there is a FW block just to be safe (it's potentially an obvious route to a fix). If there's a NGFW in front of it that does app identification it could have miscategorized the traffic. If there isn't a  block, can you do a wget for that URL from the SO box? Does that produce the same result?

**Reply by ITgeevoSOC** [reply]
Thanks for the help.

We have tested with wget command and we have connected to URL. Please note the output below.
Is the issue related to hardware? We have 8GB RAM and minimum for security onion standalone is 16GB RAM. 

![Capture](https://user-images.githubusercontent.com/90396548/135458786-f9cc2787-967e-4221-81b8-7b12ce40c558.PNG)

**Reply by clairmont32** [reply]
ok so the network connectivity appears to be there then. i cant say i recommend not exceeding recommended requirements for even POC tests for any software solution since it typically implies instability and unexpected behavior. knowing that the elk stack is behind security onion, which is java based, you're likely hitting that memory cap pretty quickly. 

you may want to do a standalone with at at least 32gb ram just to ensure that you're exceeding hardware requirements. i'd recommend calculating how much bandwidth you're expecting to monitor (pre-bpf) and allocate roughly 2 CPU per 150mbps (personally, i prefer a bit of a buffer on hardware to ensure it's not pegged at any point) but the docs state 1 CPU per roughly 200mpbs https://docs.securityonion.net/en/latest/hardware.html#cpu

**Reply by ITgeevoSOC** [reply]
Thanks for the help. I will upgrade the RAM to 32GB and check again.

**Reply by ITgeevoSOC** [reply]
We have upgrade the security onion but the issue remain. We check go to Suricata diagnostic logs (sudo /opt/so/log/suricata/suricata.log) and found the below error. We need to create an all.rules file?

5/10/2021 -- 08:34:51 - <Warning> - [ERRCODE: SC_ERR_NO_RULES(42)] - No rule files match the pattern /etc/suricata/rules/all.rules
5/10/2021 -- 08:34:51 - <Warning> - [ERRCODE: SC_ERR_NO_RULES_LOADED(43)] - 1 rule files specified, but no rules were loaded!

**Comment by dougburks** [comment]
`so-rule-update` works fine here:
```
2021-10-06 11:54:57,499 - <INFO> - Loading ./rulecat.conf.
2021-10-06 11:54:57,502 - <INFO> - Forcing Suricata version to 6.0.
2021-10-06 11:54:57,502 - <INFO> - Fetching https://rules.emergingthreats.net/open/suricata-6.0.0/emerging.rules.tar.gz.
 100% - 3035877/3035877               
2021-10-06 11:54:57,850 - <INFO> - Done.
2021-10-06 11:54:57,961 - <INFO> - Ignoring file rules/emerging-deleted.rules
2021-10-06 11:54:57,961 - <INFO> - Loading local file /opt/so/rules/nids/local.rules
2021-10-06 11:55:00,703 - <INFO> - Loaded 30602 rules.
2021-10-06 11:55:00,711 - <INFO> - Disabled 0 rules.
2021-10-06 11:55:00,711 - <INFO> - Enabled 0 rules.
2021-10-06 11:55:00,711 - <INFO> - Modified 0 rules.
2021-10-06 11:55:00,711 - <INFO> - Dropped 0 rules.
2021-10-06 11:55:00,882 - <INFO> - Enabled 130 rules for flowbit dependencies.
2021-10-06 11:55:04,565 - <INFO> - Writing rules to /opt/so/rules/nids/all.rules: total: 30602; enabled: 23301; added: 0; removed 0; modified: 0
2021-10-06 11:55:04,863 - <INFO> - Done.
```

Above you manually tested downloading `emerging.rules.tar.gz.md5`, but have you tried manually downloading `emerging.rules.tar.gz`? It's possible that you have a firewall or other device that is blocking due to something it sees in `emerging.rules.tar.gz`.

**Reply by ITgeevoSOC** [reply]
We can download manually the emerging.rules.tar.gz.md5. I don't think the issue is with firewall. Is possible to run a repair command?  

![image](https://user-images.githubusercontent.com/90396548/136379793-8c24e016-2063-4b78-845c-81859883b69e.png)

**Reply by dougburks** [reply]
> We can download manually the emerging.rules.tar.gz.md5.

To be clear, I'm not asking about `emerging.rules.tar.gz.md5`, but instead the actual `emerging.rules.tar.gz`.

Are you able to wget or curl `emerging.rules.tar.gz` from the command line of your Security Onion box and retrieve the full file with no errors?

You could also try performing a fresh installation (perhaps just in a VM) and see if it performs the same way in your network.

**Comment by ITgeevoSOC** [comment]
Yes we also can connect to  emerging.rules.tar.gz using the wget command and from browsers. Please note the screenshot below.

![image](https://user-images.githubusercontent.com/90396548/136788379-fd469236-2a40-4b79-8b9c-d9c7ba140265.png)

![image](https://user-images.githubusercontent.com/90396548/136789156-1daa2bd6-f6bd-494f-8dc5-2e784ebf626c.png)

**Comment by dougburks** [comment]
Per my previous reply, please try performing a fresh installation (perhaps just in a VM) and see if it performs the same way in your network.
